### DISCUSSION TITLE
Keyframes and Layers Issues/Ideas

### DISCUSSION THREAD

**Discussion by emmettcrass** [keiwando/toonsquid#122] [Ideas]
I do not know if this is a bug or if it just isn't implemented yet.

1. When elongating or shortening frame exposure, it would be helpful if the keyframes moved with it so they would not need to be manually moved to where they were.
2. Sometimes I have a lot of keyframes on a layer and would like to select them all, then use the Curve editor to set them all to Hold. Currently, I have to select them one-by-one to make sure they are all changed. If I try to do it by selecting multiple, then at best it will only change some of them (1-3 of them out of 10 maybe).
3. When making a new layer inside an existing layer (in the case of wanting to make lineart or color fills when a layer is animated with keyframes but is currently only the sketch), the animation will not carry over from the original layer to the others, even though it is within the same Animation layer. It would make sense to me that if they are all in the same animation layer on the timeline then they would follow the same animation.

**Comment by keiwando** [comment]
Hi!

1. You mean you want the keyframes to scale when you're extending or shortening a drawing so that the keyframe animation inside is effectively sped up / slowed down proportionally, right?
2. This is currently not implemented yet, but is already on my list. I'll bump the priority!
3. The animation layers themselves have no properties that can be keyframed. The keyframes always belong to a drawing layer, so they will not automatically reflect on other layers in the same drawing. Each one of those can have their own independent animation.
What you can do instead is to perform the coloring in a separate animation layer and then in the [transform hierarchy](https://toonsquid.com/handbook/transform/hierarchy/) set the color layer as the child of your lineart layer. Now, if you add keyframes to the drawing layers containing your lineart, the color layer should automatically follow.
